olp-cpp-sdk-authentication/include/olp/authentication/AuthenticationSettings.h

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <boost/optional.hpp>
+#include <memory>
+#include <string>
+
+#include <olp/core/http/NetworkProxySettings.h>
+
+#include "AuthenticationApi.h"
+
+namespace olp {
+namespace http {
+class Network;
+}
+
+namespace thread {
+class TaskScheduler;
+}
+
+
+namespace authentication {
+static const std::string kHereAccountProductionTokenUrl =
+    "https://account.api.here.com/oauth2/token";
+
+/**
+ * @brief Configures the `TokenEndpoint` instance.
+ *
+ * Contains settings that customize the `TokenEndpoint` class.
+ */
+struct AUTHENTICATION_API AuthenticationSettings {
+  /**
+   * @brief The network instance that is used to internally operate with the OLP
+   * services.
+   */
+  std::shared_ptr<http::Network> network_request_handler;
+
+  /**
+   * @brief (Optional) The `TaskScheduler` class that is used to manage
+   * the callbacks enqueue.
+   */
+  std::shared_ptr<thread::TaskScheduler> task_scheduler;
+
+  /**
+   * @brief (Optional) The configuration settings for the network layer.
+   */
+  boost::optional<http::NetworkProxySettings> network_proxy_settings;
+
+  /**
+   * @brief (Optional) The server URL of the token endpoint.
```

Comment:
```suggestion
   * @brief The server URL of the token endpoint.
```
Would it make sense to use the `boost::optional<std::string>` type for the property instead?

Reply:
Not sure about this one.
`token_endpoint_url` has default value. What is expected for `boost::none` then?

Reply:
If it has a default value, it's not an optional property anymore (as it always has a value), is it?

Reply:
I don't think we should create the impression that this value is optional. Actually, without the user providing this value, we cannot do authentication. I would say a non-optional value with a default assigned URL is a good strong message :)

Reply:
Yeah, sorry, I misunderstood the original comment.